### PR TITLE
Upgrade version of p2-feature to 9.2.0.

### PR DESCRIPTION
**Pull request by nikhilnanivadekar**

### CODE REVIEW

**Comment by nikhilnanivadekar** [comment]
@ujhelyiz , bergmanngabor , can either of you please review?
@ujhelyiz this command `mvn org.eclipse.tycho:tycho-versions-plugin:set-version -DnewVersion=9.2.0` keeps on failing. I ended up doing a find and replace. 
```
[ERROR] Internal error: java.lang.RuntimeException: Failed to load p2 repository with ID 'collections.plugin.repository' from location file://C:/EclipseCollections/eclipse-collections/p2-feature/org.eclipse.collections.feature/../../p2-repository/org.eclipse.collections/target/repository: URI has an authority component -> [Help 1]
org.apache.maven.InternalErrorException: Internal error: java.lang.RuntimeException: Failed to load p2 repository with ID 'collections.plugin.repository' from location file://C:/EclipseCollections/eclipse-collections/p2-feature/org.eclipse.collections.feature/../../p2-repository/org.eclipse.collections/target/repository
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:121)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
```
If this looks ok, after it is merged, I will create another milestone release and also upgrade the sim rel contribution.
Thank you very much for your help!

**Comment by ujhelyiz** [comment]
I don't think the files in this change are fine, because the version numbers in `p2-feature/org.eclipse.collections.feature/feature.xml` and `p2-feature/org.eclipse.collections.feature/feature.xml` are not in synch. I looked at the travis build output, and it doesn't seem to me that it tried to build this repository.

Furthermore, for this project group I would always use a -SNAPSHOT version in the pom.xml files, even for releases. This is because Eclipse plug-ins and features does not have any notion for snapshot and release builds.

----

About the issue with set-version, at first glance the suggested URI seems to be fine by me, but I consulted with one of my collegues about his experiences with similar issues in Windows. He said, the respective URL might or might not be fine, as various tools expect to have a various number of `/` symbols after the `file:` prefix. You could explicitly override this url by overriding the [collections.repository.url property](https://github.com/nikhilnanivadekar/eclipse-collections/blob/master/p2-feature/pom.xml#L40) for the build.

But before I start using this property, I'd look at the mentioned location to check whether there is the generated repository of the original build containing the plug-in.

File: p2-feature/org.eclipse.collections.feature/pom.xml [hunk 1]
```diff
@@ -15,7 +15,7 @@
 	<parent>
 		<groupId>org.eclipse.collections</groupId>
 		<artifactId>p2-feature</artifactId>
-		<version>9.1.0-SNAPSHOT</version>
+		<version>9.2.0</version>
```
Comment (ujhelyiz) [inline p2-feature/org.eclipse.collections.feature/pom.xml]: This has to be synchronized with the version number in feature.xml and the category.xml file version in the repository project.

File: p2-feature/pom.xml [hunk 2]
```diff
@@ -12,7 +12,7 @@
     <modelVersion>4.0.0</modelVersion>
 
     <groupId>org.eclipse.collections</groupId>
-    <version>9.1.0-SNAPSHOT</version>
+    <version>9.2.0</version>
```
Comment (ujhelyiz) [inline p2-feature/pom.xml]: It's better to keep the snapshot versions in these projects.

**Comment by nikhilnanivadekar** [comment]
@ujhelyiz Thanks for the review. I have fixed all the points which you pointed out. Can you please review again?
It needs 3 "/"s after `file:` for the command to work. Thanks for the tip.

**Comment by nikhilnanivadekar** [comment]
Merging based on review from @ujhelyiz 
IP Validation is not running but verified that the commit is signed-off and my ECA status has not changed.

**Pull request merged**